go/vt/vttablet/tabletmanager/vreplication/table_plan_builder.go

```diff
@@ -629,7 +631,7 @@ func (tpb *tablePlanBuilder) analyzeExtraSourcePkCols(colInfos []*ColumnInfo, so
 			if !col.IsGenerated {
 				// We shouldn't get here in any normal scenario. If a column is part of colInfos,
 				// then it must also exist in tpb.colExprs.
-				return fmt.Errorf("column %s not found in table expressions", col.Name)
+				return vterrors.Errorf(vtrpc.Code_INTERNAL, "column %s not found in table expressions", col.Name)
```

Comment:
Better to use another code: https://github.com/vitessio/vitess/blob/8575b170183bce389103d451965dc12070f4e36d/proto/vtrpc.proto#L74-L127

One that reflects the general case. I think FAILED_PRECONDITION might be the right one. Although NOT_FOUND would also fit.